api/runnerpool/placer_tracker.go

```diff
@@ -113,7 +113,22 @@ func (tr *placerTracker) HandleDone() {
 
 // RetryAllBackoff blocks until it is time to try the runner list again. Returns
 // false if the placer should stop trying.
-func (tr *placerTracker) RetryAllBackoff(numOfRunners int) bool {
+func (tr *placerTracker) RetryAllBackoff(numOfRunners int, err error) bool {
+	// If there are no runners and last call to provision runners failed due
+	// to a user error (or misconfiguration) then fail fast.
+	if numOfRunners == 0 && err != nil {
+		// IsFuncError currently synonymous with tag: 'blame == user'
+		// See: runner_fninvoke.handleFnInvokeCall2
+		if models.IsFuncError(err) {
+			// We also sanity check for a 502 before returning.
```

Comment:
minor nit: do we need this extra check here? I think IsFuncError() should be good enough.

Reply:
Possibly not; it's hard to envisage a user error that prevents us from scaling up (from zero) where we *don't* want to bail immediately. On the other hand, I'd rather err on the side of being overly cautious in the first instance and then relax this condition.